src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java

```diff
@@ -529,6 +530,7 @@ public void setGuardPos(final BlockPos guardPos)
     @Override
     public BlockPos getPositionToFollow()
     {
+        PlayerEntity followPlayer = getPlayerFromUUID(followPlayerUUID, this.colony.getWorld());
         if (getSetting(GUARD_TASK).getValue().equals(GuardTaskSetting.FOLLOW) && followPlayer != null)
```

Comment:
In general it's fine. But we should have another check here to make sure the player is in the same dimension.

Reply:
While working on this I found out that following guards can't use portals so they wouldn't be able to follow players across dimensions anyway, is this intended?

Reply:
Yes fully, citizens don't work well outside of their own dimension.